crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -1676,6 +1683,86 @@ impl<'db> From<ClassLiteralType<'db>> for Type<'db> {
     }
 }
 
+/// Representation of a single `Protocol` class definition.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(super) struct ProtocolClassLiteral<'db>(ClassLiteralType<'db>);
+
+impl<'db> ProtocolClassLiteral<'db> {
+    pub(super) fn members(self, db: &'db dyn Db) -> &'db ordermap::set::Slice<Name> {
```

Comment:
Keep in mind (both in terms of method naming, and where you put what functionality) that autocomplete is definitely going to require the ability for `ClassLiteralType` itself to have a "list all members" method. And if  `ProtocolClassLiteral` will be used inside a future `Type::ProtocolInstance`, it will probably also need to expose that list-all-members method, without all the Protocol-specific filtering.  That could suggest some future refactoring to share some aspects of list-members functionality, but more concretely I think it suggests that we should reserve the method name `members` for the general "list all known members" that all types will need, and use something like `protocol_members` to distinguish this protocol-specific method.

Reply:
Right, makes sense. It seems like also the filtering-out of some "special" names is something we might not want to do for autocomplete?